lib/events/kubernetesDeploy.ts

```diff
@@ -0,0 +1,214 @@
+        logger.warn(`Received event had no SdmGoal`);
+        return Success;
+    }
+
+    return Promise.all(ef.data.SdmGoal.map(async g => {
+        const goalEvent = g as SdmGoalEvent;
+
+        const progressLog = new WriteToAllProgressLog(goalEvent.name, new LoggingProgressLog(goalEvent.name, "debug"),
+            await params.configuration.sdm.logFactory(context, goalEvent));
+
+        const eligible = await eligibleDeployGoal(goalEvent, params);
+        if (!eligible) {
+            llog("SDM goal event is not eligible for Kubernetes deploy", logger.info, progressLog);
+            return Success;
+        }
+
+        let app: KubernetesApplication;
+        try {
+            app = getKubernetesGoalEventData(goalEvent);
+        } catch (e) {
+            e.message = `Invalid SDM goal event data: ${e.message}`;
+            llog(e.message, logger.error, progressLog);
+            throw e;
+        }
+        if (!app) {
+            llog("SDM goal event has no Kubernetes application data", logger.info, progressLog);
+            return Success;
+        }
+        const appId = deployAppId(goalEvent, context, app);
+
+        try {
+            const result = await deployApplication(goalEvent, context);
+
+            const updateParams: UpdateSdmGoalParams = {
+                state: (result.code) ? SdmGoalState.failure : SdmGoalState.success,
+                description: result.description,
+                error: (result.code) ? new Error(result.message) : undefined,
+                externalUrls: result.externalUrls,
+            };
+            try {
+                await updateGoal(context, goalEvent, updateParams);
+            } catch (e) {
+                const msg = `Failed to update SDM goal '${stringify(goalEvent)}' with params ` +
+                    `'${stringify(updateParams)}': ${e.message}`;
+                llog(msg, logger.error, progressLog);
+                result.message = `${e.message}; ${msg}`;
+            }
+            if (!result.code) {
+                result.code = 0;
+            }
+            return result as ExecuteGoalResult & HandlerResult;
+        } catch (e) {
+            const msg = `Failed to deploy ${appId}: ${e.message}`;
+            return failGoal(context, goalEvent, msg, progressLog);
+        }
+    }))
+        .then(reduceResults);
+};
+
+/**
+ * Create an event handler registration for this SDM to deploy
+ * requested Kubernetes applications.
+ */
+export function kubernetesDeployHandler(self: string)
+    : EventHandlerRegistration<KubernetesDeployRequestedSdmGoal.Subscription, KubernetesDeployParameters> {
+    return {
+        name: "KubernetesDeploy",
+        description: "Deploy application resources to Kubernetes cluster",
+        tags: ["deploy", "kubernetes"],
+        subscription: GraphQL.subscription({ name: "KubernetesDeployRequestedSdmGoal", variables: { fulfillmentName: self } }),
+        paramsMaker: KubernetesDeployParameters,
```

Comment:
That reminds me that I failed to deprecate `paramsMaker` for event handlers. Will have to do that too.

Reply:
What should be used?

Reply:
You can’t use anything else yet. That comment was mostly for myself to remind to add the parameters stuff to event handlers.